safe-global/safe-wallet-web
# feat: Recovery tracking

## What it solves

Resolves #2835 

## How this PR fixes it

- Adds recovery events and documents them ([Notion](https://www.notion.so/safe-global/Google-Analytics-custom-events-bfd072d2bb704394860a59724dc3c66f?pvs=4#5f9ae8c0a430458895cee38e5622f43f))
- Extends the `useTxStepper` hook to receive an optional argument for `eventCategory` and track Next/Back clicks

## ToDos

- [x] The `recovery_attempt` event is sent before the user submits the transaction in their wallet
- [x] No event is sent if the expired recovery tx is canceled

## How to test it

Open a Safe, go through the list of events and make sure they are sent as described

## Checklist
